src/processor/operator/persistent/reader/csv/parallel_csv_reader.cpp

```diff
@@ -182,11 +189,24 @@ static std::unique_ptr<TableFuncLocalState> initLocalState(TableFunctionInitInpu
     return localState;
 }
 
+static double progressFunc(TableFuncSharedState* sharedState) {
+    auto state = reinterpret_cast<ParallelCSVScanSharedState*>(sharedState);
+    if (state->totalSize == 0) {
+        return 0.0;
+    } else if ((state->blockIdx * CopyConstants::PARALLEL_BLOCK_SIZE) * (state->fileIdx + 1) >=
```

Comment:
Do we have the guarantee that all files will have the same block? I don't quite understand why we do `* (state->fileIdx + 1)`.

You probably have a wrong understanding of parallel csv reader. We don't parallelize by allocating each thread to a file. Instead, we parallelize by assigning all threads to one file and proceed to the next only when current file has finished.

So you could do this in a similar fashion as serial csv reader, i.e. track numBlocksReadByFile in shared state.